IndicoV2/Storage/StorageClient.cs

```diff
@@ -63,25 +63,32 @@ public JArray Serialize(IEnumerable<IFileMetadata> filesMetadata)
 
         public async Task<(string Name, string Meta)> UploadAsync(Stream content,
             string filePath,
-            CancellationToken cancellationToken) =>
-            (await UploadAsync(new[] {(filePath, content)}, cancellationToken)).SingleOrDefault();
+            CancellationToken cancellationToken, int batchSize = 20) =>
+            (await UploadAsync(new[] { (filePath, content) }, cancellationToken, batchSize)).SingleOrDefault();
 
         public async Task<(string Name, string Meta)[]> UploadAsync(IEnumerable<(string Path, Stream Content)> files,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken, int batchSize = 20)
         {
-            var content = await CreateRequest(files, cancellationToken);
-            var response = await _indicoClient.HttpClient.PostAsync(UploadUri, content, cancellationToken);
-
-            using (var reader = new JsonTextReader(new StreamReader(await response.Content.ReadAsStreamAsync())))
+            var results = new (string Name, string Meta)[] { };
```

Comment:
Let me know how this looks for batching

Reply:
This looks good! One question, what happens if I put in edge case values for batch size? Like 0 or -1?

Reply:
ah good point. I added argumentexception for invalid batch sizes